wasmplugin/plugin.go

```diff
@@ -0,0 +1,363 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package wasmplugin
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3"
+	ctypes "github.com/corazawaf/coraza/v3/types"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+type VMContext struct {
```

Comment:
Because `corazaPlugin` is private, it's not a big deal, but how about also keeping this private and just adding `func NewVMContext() types.VMContext { return &vmContext }`?

Reply:
Sounds good. I'll change

Reply:
Done